File: warehouse_management/overrides/warehouse.py
from erpnext.stock.doctype.warehouse.warehouse import Warehouse
import frappe
from frappe import _


class CustomWarehouse(Warehouse):

    def validate(self):
        super().validate()
        self.validate_warehouse_price_flags()

    def validate_warehouse_price_flags(self):
        if self.custom_show_warehouse_pricing and self.custom_show_wholesale_price:
            frappe.throw(
                _("Only one of Warehouse Pricing or Wholesale Price can be selected.")
            )

    def autoname(self):
        if not self.custom_warehouse_code:
            frappe.throw("Custom Warehouse Code is required")
        if self.custom_show_warehouse_pricing and self.custom_show_wholesale_price:
            frappe.throw(
                _("Only one of Warehouse Pricing or Wholesale Price can be selected.")
            )
        self.name = self.custom_warehouse_code
        frappe.msgprint(self.name)

File: warehouse_management/public/warehouse.js
frappe.ui.form.on('Warehouse', {
    custom_show_warehouse_pricing(frm) {
        if (frm.doc.custom_show_warehouse_pricing) {
            frm.set_value('custom_show_wholesale_price', 0);
        }
    },

    custom_show_wholesale_price(frm) {
        if (frm.doc.custom_show_wholesale_price) {
            frm.set_value('custom_show_warehouse_pricing', 0);
        }
    },

    validate(frm) {
        if (
            frm.doc.custom_show_warehouse_pricing &&
            frm.doc.custom_show_wholesale_price
        ) {
            frappe.throw(__('Only one of Warehouse Pricing or Wholesale Price can be selected.'));
        }
    }
});
